### ISSUE TITLE
Internal Link Routing for Tags and Previous/Next Buttons Not Working Properly

### ISSUE THREAD

**Issue by caljup** [gatsbyjs/gatsby#25500]
I've had a Gatsby site now for close to a year, and after updating the CLI and React today internal linking has been acting weird. I didn't change anything else and reviewed all dependencies, but the problem persists.

Essentially for my blog I have a tags page programmatically generated from markdown files with the appropriate information queried from GraphQL. Before the update I would click on a tag and a feed of posts under that tag would be displayed that when you click a post would route you to a URL like this: www.thisisasite.com/blog-post, but now if I click a post in the feed instead it tries to navigate to www.thisisasite.com/tags/tag/blog-post which results in a 404 error.

The same thing occurs on each post where I have a Next and Previous button to switch between posts. In that case if you hit Next instead of going to www.thisisasite.com/blog-post-2 it tries to navigate to www.thisisasite.com/blog-post-1/blog-post-2.

Normal site directory is set up with pages, templates, and styles all in separate folders underneath. The markdown posts are located in a folder under the pages folder. 

When I query the slug I set up routing according to documentation with: <Link to ={slug}>, but it's still giving me the above error.

Under gatsby-node.js I have everything set up exactly based on the documentation here: https://www.gatsbyjs.org/docs/adding-tags-and-categories-to-blog-posts/

This is the code for creating each page:

        //Create single page posts
        posts.forEach(({node}, index) => {
            createPage({
                path: node.fields.slug,
                component: templates.singlePost,
                context: {
                    // Passing slug for template to use to get post
                    slug: node.fields.slug,
                    prev: index === 0 ? null : posts[index - 1].node,
                    next: index === (posts.length - 1) ? null : posts[index + 1].node
                },
            })
        })

        //Get all tags
        let tags = []
        _.forEach(posts, edge => {
            if (_.get(edge, 'node.frontmatter.tags')) {
                tags = tags.concat(edge.node.frontmatter.tags)
            }
        })

        //Removes duplicate tags
        tags = _.uniq(tags)

        //Create Pages for Tag Posts
        tags.forEach(tag => {
            createPage({
                path: `/tag/${slugify(tag)}`,
                component: templates.tagPosts,
                context: {
                    tag,
                },
            })
        })
    })
}

Config file is set as:

    `gatsby-plugin-react-helmet`,
    `gatsby-plugin-sass`,
    `gatsby-transformer-remark`,
    `gatsby-plugin-catch-links`,
    `gatsby-plugin-remove-trailing-slashes`,
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `images`,
        path: `${__dirname}/src/images/`,
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `pages`,
        path: `${__dirname}/src/pages/`,
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `markdown-posts`,
        path: `${__dirname}/src/pages/markdown-posts/`,
      },
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `fonts`,
        path: `${__dirname}/src/styles/fonts/`,
      },
    },
    `gatsby-transformer-sharp`,
    `gatsby-plugin-sharp`,
    `gatsby-background-image`,
    `gatsby-plugin-styled-components`,

My inclination is that after the update the pre-rendering for paths with <Link> is bugging out and trying to create paths for posts on pages using the current directory or component as the root instead of the main domain.

Any help much appreciated or if y'all have seen anything like this happening since the update.

**Comment by LekoArts** [comment]
Hi, thanks for the issue!

We recently fixed a broken behavior: https://github.com/gatsbyjs/gatsby/pull/24054
TL;DR: Link now also supports relative links.

Can you please share a reproduction of your problem, a link to your repo, or check whether your links all begin with a `/` or not? Because if not they are all relative.

**Comment by caljup** [comment]
Added in the linking with / and that fixed the problem. Since Link now supports relative links, links to internal pages from current route will treat the current directory as having a trailing slash, which was why www.thisisasite.com/blog-post would set the link to the next post as www.thisisasite.com/blog-post/blog-post 2.

Solutution was easy enough. If you have a link going to programatically generated slugs and you are passing 'slug' into the component code previously looked like this

                <Link to = {slug}}>
                    {title}
                </Link>

Just changed link to {`/${slug}`} and that fixed the issue

                <Link to = {`/${slug}`}>
                    {title}
                </Link>

Thanks for your help @LekoArts
